src/plots/plots.js

```diff
@@ -2364,14 +2364,15 @@ plots.doCalcdata = function(gd, traces) {
     // clear stuff that should recomputed in 'regular' loop
     if(hasCalcTransform) clearAxesCalc(axList);
 
-    // 'regular' loop
-    for(i = 0; i < fullData.length; i++) {
-        var cd = [];
-
+    function calci(i, isContainer) {
         trace = fullData[i];
+        _module = trace._module;
+
+        if(!!_module.isContainer !== isContainer) return;
```

Comment:
Ha. Nice fix. I wonder if other carpet-dependent blocks (e.g. [here](https://github.com/plotly/plotly.js/blob/3d65de3adb20ca0c75e9ecb67739c5d801cc3b24/src/plots/plots.js#L948-L963)) could be simplified (or at least generatlized) using `_module.isContainer`?

Reply:
Probably some such blocks could... though I don't see much to do with that `carpetDependents` loop. It might be nice to move it out of plots/plots and into the `carpet` module, but probably not in this PR.